modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/outgoing/OutgoingSnapshotsManager.java

```diff
@@ -0,0 +1,145 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed.raft.snapshot.outgoing;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import org.apache.ignite.internal.logger.IgniteLogger;
+import org.apache.ignite.internal.logger.Loggers;
+import org.apache.ignite.internal.manager.IgniteComponent;
+import org.apache.ignite.internal.table.message.SnapshotMetaRequest;
+import org.apache.ignite.internal.table.message.SnapshotMvDataRequest;
+import org.apache.ignite.internal.table.message.SnapshotRequestMessage;
+import org.apache.ignite.internal.table.message.SnapshotTxDataRequest;
+import org.apache.ignite.internal.table.message.TableMessageGroup;
+import org.apache.ignite.network.MessagingService;
+import org.apache.ignite.network.NetworkAddress;
+import org.apache.ignite.network.NetworkMessage;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Outgoing snapshots manager.
+ */
+public class OutgoingSnapshotsManager implements IgniteComponent {
+    /** Logger. */
+    private static final IgniteLogger LOG = Loggers.forClass(OutgoingSnapshotsManager.class);
+
+    /** Messaging service. */
+    private final MessagingService messagingService;
+
+    /** Map with outgoing snapshots. */
+    private final ConcurrentMap<UUID, OutgoingSnapshot> outgoingSnapshots = new ConcurrentHashMap<>();
+
+    /**
+     * Constructor.
+     *
+     * @param messagingService Messaging service.
+     */
+    public OutgoingSnapshotsManager(MessagingService messagingService) {
+        this.messagingService = messagingService;
+    }
+
+    /**
+     * Returns a messagins service.
+     */
+    public MessagingService messagingService() {
+        return messagingService;
+    }
+
+    @Override
+    public void start() {
+        messagingService.addMessageHandler(TableMessageGroup.class, this::messageHandler);
+    }
+
+    @Override
+    public void stop() throws Exception {
+    }
+
+    /**
+     * Registers an outgoing snapshot in the manager.
+     *
+     * @param snapshotId Snapshot id.
+     * @param outgoingSnapshot Outgoing snapshot.
+     */
+    void startOutgoingSnapshot(UUID snapshotId, OutgoingSnapshot outgoingSnapshot) {
```

Comment:
Should it be called `registerOutgoiingSnapshot`?

Reply:
Maybe